### ISSUE TITLE
Make the command nvda+k to report overwritten link text on first press

### ISSUE THREAD

**Issue by Adriani90** [nvaccess/nvda#16415]
### Is your feature request related to a problem? Please describe.
1. Take this code example and paste it into the address bar of the browser
```
data:text/html,<div> <body><strong>Useful links</strong><br><a href="blank" download="SwitchAudioFileConverter.exe" aria-label="opens in a new tab">Windows Download</a><br> <a href="../components/switchmaci.zip">Mac Download</a><a href="blank" download="SwitchAudioFileConverter.exe" aria-label="Android download">Android Download</a></body></div>
```
Use case in the wild, this website contains lots of such links, making it almost impossible to work with it accessibly:
https://www.nch.com.au/switch/index.html
2. Try to read all links from the code snippet above in browse and focus mode, by pressing nvda+tab, nvda+shift+o, by using the mouse or by using object navigation.

### Actual behavior:
NVDA reports only aria-label "opens in a new tab" on the link with the link text "Windows download", ignoring the link text. There is no way to read the actual link text unless you use the mouse or object navigation.

### Describe the solution you'd like
Make NVDA+k command to report html link text on first press.

### Describe alternatives you've considered
None

### Additional context
In #16363 I proposed to honnor link text instad of aria label but this is not what standard web design agreed on. Given we cannot mittigate all the bad authoring issues, in this case there should be at least an easier way to read the actual link text.

**Comment by Adriani90** [comment]
cc: @XLTechie

**Comment by CyrilleB79** [comment]
What would you expect in your example? And on which link?

**Comment by Adriani90** [comment]
On every link, I expect to hear the html link text when pressing nvda+k. Many links on that website as well as the first link in my code example, have the aria label "opens in a new tab" which is reported when navigating to the links because aria label is honored by screen readers in general over the html link text.
However, the actual link text e.g. for the first link in the code example is "Windows Download" and this is not reported when navigating to the link.
See comments in #16163 for more details and what NV Access expects as well.

**Comment by seanbudd** [comment]
This may make more sense to be displayed in the browsable message generated by pressing nvda+k twice

**Comment by Adriani90** [comment]
Sean not really, because on a such a bad authored website users need the link text as soon as possible while navigating. It would be horrible if users have to always have to open a browseable window on every of the links with wrong aria label in order to hear what the link is really about.Von meinem iPhone gesendetAm 23.04.2024 um 02:22 schrieb Sean Budd ***@***.***>:﻿
This may make more sense to be displayed in the browsable message generated by pressing nvda+k twice

**Comment by michaelDCurran** [comment]
@Adriani90  Adding this to NVDA+k single press may cause ambiguity or confusion. Not to mention less efficient (presuming the extra info is spoken before the URL).
Adding it to NVDA+k browsable will provide a way to clearly delineate the URL from the text.
E.g.:
```
URL: https://github.com/michaeldcurran
Text: @michaelDCurran 
```

I'd also like to understand your proposed way of detecting when the link's text should be presented, verses when it should not. Not that in many cases where aria-label is used, the links text might be just something like `&mbsp;` or ` `  or, it might be rather meaningful like `download` as you pointed out. 
I agree it would be nice to provide some way for the user to get to this text, but it should not confuse or slow down an existing feature. thus why I propose it to be in the browseable message.

**Comment by Adriani90** [comment]
I agree in majority, I think I should have been clearer in formulating this description. I had in mind a solution where nvda+k has three presses.first = link textSecond = URLThird = browseable message.But yeah it might lead to controversial discussions, browseable message will be indeed a good compromise.Von meinem iPhone gesendetAm 23.04.2024 um 08:40 schrieb Michael Curran ***@***.***>:﻿
Using aria label to convey "opens in a new tab" is extremely poor and invalid use of ARIA. The author should be using aria-description or aria-describedby. Screen readers cannot work around every possible author error. But I feel that the balance would be  placing it in the browseable message.

**Comment by Adriani90** [comment]
So if a browseable message is the way to go here, then in my view the NVDA+k command should only display a browseable message on the first press, and present both link text and URL in it.
Currently, the browseable message is displayed on nvda+k twice.

**Comment by Adriani90** [comment]
I also found #10783, it might also be a good option. Fixing that would be Better than the current behavior at least.
